examples/stress_tests/particles.rs

```diff
@@ -0,0 +1,226 @@
+                title: "GGRS particles stress test".to_owned(),
+                ..default()
+            }),
+            ..default()
+        }))
+        .add_systems(Startup, setup)
+        .add_systems(Startup, spawn_particles) // spawn an initial burst of particles
+        .add_systems(
+            GgrsSchedule,
+            (
+                spawn_particles.run_if(spawn_pressed),
+                update_particles,
+                despawn_particles,
+            ),
+        )
+        .insert_resource(Session::P2P(session))
+        .insert_resource(ClearColor(Color::BLACK))
+        .insert_resource(ParticleRng(GameRng::seed_from_u64(123)))
+        .add_systems(Update, print_events_system)
+        .run();
+
+    Ok(())
+}
+
+fn setup(mut commands: Commands) {
+    commands.spawn(Camera2dBundle::default());
+}
+
+fn spawn_pressed(inputs: Res<PlayerInputs<Config>>) -> bool {
+    inputs.iter().any(|(i, _)| *i & INPUT_SPAWN != 0)
+}
+
+fn spawn_particles(mut commands: Commands, args: Res<Args>, mut rng: ResMut<ParticleRng>) {
+    let s = 200.0;
+    let ttl = args.fps * 5;
+
+    for _ in 0..args.rate {
+        commands
+            .spawn((
+                SpriteBundle {
+                    sprite: Sprite {
+                        color: Color::ORANGE,
+                        custom_size: Some(Vec2::splat(5.0)),
+                        ..default()
+                    },
+                    ..default()
+                },
+                Velocity(vec3(rng.gen_range(-s..s), rng.gen_range(-s..s), 0.0)),
+                Ttl(ttl),
+            ))
+            .add_rollback();
+    }
+}
+
+fn update_particles(mut particles: Query<(&mut Transform, &mut Velocity)>, args: Res<Args>) {
+    let time_step = 1.0 / args.fps as f32; // todo: replace with bevy_ggrs resource?
+    let gravity = Vec3::NEG_Y * 200.0;
+
+    for (mut transform, mut velocity) in &mut particles {
+        **velocity += gravity * time_step;
+        transform.translation += **velocity * time_step;
+    }
+}
+
+fn despawn_particles(mut commands: Commands, mut particles: Query<(Entity, &mut Ttl)>) {
+    for (entity, mut ttl) in &mut particles {
+        **ttl -= 1;
+        if **ttl == 0 {
+            commands.entity(entity).despawn_recursive();
+        }
+    }
+}
+
+fn print_events_system(mut session: ResMut<Session<Config>>) {
+    match session.as_mut() {
+        Session::P2P(s) => {
+            for event in s.events() {
+                match event {
+                    GgrsEvent::Disconnected { .. } | GgrsEvent::NetworkInterrupted { .. } => {
+                        warn!("GGRS event: {event:?}")
+                    }
+                    GgrsEvent::DesyncDetected {
+                        local_checksum,
+                        remote_checksum,
+                        frame,
+                        ..
+                    } => panic!("Desync on frame {frame}. Local checksum: {local_checksum:X}, remote checksum: {remote_checksum:X}"),
```

Comment:
I think having a command line option to choose between `error!` and `panic!` on desync would be nice. Switching this to `error!`, I noticed that while the checksums differ, the actual simulations look identical, so whatever desync there is, it's not noticeable without the checksum.

Reply:
Good idea. I noticed I changed this back and forth a number of times when debugging.

And good spot about the visual similarity.